modules/python/package/cv2/_extra_py_code/__init__.py

```diff
@@ -1,53 +1,65 @@
+import os
 import sys
 import importlib
 
 __all__ = ['init']
 
-
 DEBUG = False
 if hasattr(sys, 'OpenCV_LOADER_DEBUG'):
     DEBUG = True
 
 
-def _load_py_code(base, name):
+def _load_extra_py_code_for_module(base, name):
+    module_name = "{}.{}".format(__name__, name)
+    export_module_name = "{}.{}".format(base, name)
     try:
-        m = importlib.import_module(__name__ + name)
+        m = importlib.import_module(module_name)
     except ImportError:
-        return  # extension doesn't exist?
+        # Extension doesn't contain extra py code
+        return False
 
-    if DEBUG: print('OpenCV loader: added python code extension for: ' + name)
+    if DEBUG:
+        print('OpenCV loader: added python code extension for: ' + name)
 
     if hasattr(m, '__all__'):
-        export_members = { k : getattr(m, k) for k in m.__all__ }
+        export_members = {k: getattr(m, k) for k in m.__all__}
     else:
         export_members = m.__dict__
 
-    for k, v in export_members.items():
-        if k.startswith('_'):  # skip internals
-            continue
-        if isinstance(v, type(sys)):  # don't bring modules
-            continue
-        if DEBUG: print('    symbol: {} = {}'.format(k, v))
-        setattr(sys.modules[base + name ], k, v)
+    # If it is C extension module it is already loaded by cv2 package
+    if export_module_name in sys.modules:
+        for k, v in export_members.items():
+            if k.startswith('_'):  # skip internals
+                continue
+            if isinstance(v, type(sys)):  # don't bring modules
+                continue
+            if DEBUG: print('    symbol: {} = {}'.format(k, v))
+            setattr(sys.modules[export_module_name], k, v)
+    else:
+        # Otherwise we should add this module to modules list manually
+        if not hasattr(base, name):
+            setattr(sys.modules[base], name, m)
+        sys.modules[export_module_name] = m
+
+    del sys.modules[module_name]
```

Comment:
BTW, Any thoughs about removal of these "extra" modules?

- do we need that cleanup?
- if yes, then it should be updated. AFAIK, currently it doesn't remove imported submodules properly.